### PR TITLE
Fix mysql backend openWithoutDB ssl-mode not work

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: I3bf5ef9deae121a51f4e1b4ce8132958194bde31

### CODE REVIEW

File: hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlSessions.java [hunk 1]
```diff
@@ -183,43 +183,35 @@ protected String buildExistsTable(String table) {
...
-        if (jdbcUrl.endsWith("/")) {
-            jdbcUrl = String.format("%s%s", jdbcUrl, this.database());
-        } else {
-            jdbcUrl = String.format("%s/%s", jdbcUrl, this.database());
-        }
-        String url = new URIBuilder().setPath(jdbcUrl)
-                                     .setParameter("socketTimeout",
-                                                   String.valueOf(timeout))
-                                     .toString();
+        String url = this.buildUri(true, timeout);
```
Comment (javeme) [inline hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlSessions.java]: the origin logic is autoReconnect=false

File: hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlSessions.java [hunk 2]
```diff
@@ -183,43 +183,35 @@ protected String buildExistsTable(String table) {
      * Connect DB without specified database
      */
     protected Connection openWithoutDB(int timeout) {
-        String jdbcUrl = this.config.get(MysqlOptions.JDBC_URL);
-        String url = new URIBuilder().setPath(jdbcUrl)
-                                     .setParameter("socketTimeout",
-                                                   String.valueOf(timeout))
-                                     .toString();
+        String url = this.buildUri(true, timeout);
```
Comment (javeme) [inline hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlSessions.java]: the origin logic is autoReconnect=false and no database parameter

**Comment by I-DO-NOT-KNOW** [comment]
> Change-Id: I3bf5ef9deae121a51f4e1b4ce8132958194bde31

你好，请问下现在是怎么解决0.10.4版本openWithoutDB ssl-mode不起作用这个问题。现在初始化mysql一直报SSL这个问题。

**Comment by javeme** [comment]
@I-DO-NOT-KNOW 可以cherry-pick这个commit

**Comment by I-DO-NOT-KNOW** [comment]
> @I-DO-NOT-KNOW 可以cherry-pick这个commit

好的，谢谢

File: hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlSessions.java (hunk 2 above)
Comment (Linary) [reply hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlSessions.java]: autoReconnect really do not need, but need useSSL param

**Comment by Linary** [comment]
MySQL backend, execute `init-store.sh`
image

File: hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlSessions.java [hunk 3]
```diff
@@ -207,35 +207,39 @@ protected Connection openWithDB(int timeout) {
      * Connect DB with specified database
      */
     private Connection open(boolean autoReconnect) throws SQLException {
-        String url = this.buildUri(autoReconnect, null);
+        String url = this.buildUri(true, true, autoReconnect, null);
         return this.connect(url);
     }
 
-    protected String buildUri(boolean autoReconnect, Integer timeout) {
+    protected String buildUri(boolean withdb, boolean withConnParams,
```
Comment (javeme) [inline hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlSessions.java]: rename to withDB
and prefer to set withConnParams as first param

**Pull request merged**